src/main/java/com/redhat/cloud/policies/engine/process/Receiver.java

```diff
@@ -32,10 +32,14 @@ public class Receiver {
     @ActivateRequestContext
     public void process(String payload) {
         Log.tracef("Received payload: %s", payload);
-        payloadParser.parse(payload).ifPresent(event -> {
-            statelessSessionFactory.withSession(statelessSession -> {
-                eventProcessor.process(event);
+        try {
+            payloadParser.parse(payload).ifPresent(event -> {
+                statelessSessionFactory.withSession(statelessSession -> {
+                    eventProcessor.process(event);
+                });
             });
-        });
+        } catch (Exception e) {
+            Log.errorf(e, "Payload processing failed: %s", payload);
```

Comment:
Payload might be too big for a log message.

Reply:
Good point. I don't think that will ever be an issue for the OpenShift log, but it may cause a failure when a batch of log entries is pushed to CloudWatch.

Do you want to truncate the payload when it exceeds a chosen max length (which one)?

Reply:
Can the error contain the error message? We should leave out payload to debug data/log.

Reply:
I updated the code. The log entry depends on the log level now.

Reply:
The log entry ar `error` level will contain all useful details about the exception.

Reply:
The `e` argument which is passed to `Log.error[f]` method will trigger a whole stack trace print in the logs. That includes the exception message so there's no need to repeat it in the message passed to `Log.error[f]`.

Reply:
With the current code, the exception message and the full stack trace are printed whether the debug log is enabled or not. `e` is passed in both cases but in one case it's the first arg and in the other case it's the last arg. I don't know why they built the API like that, it can be confusing...

Reply:
With the `e` passed as second arg, it is taken as a param for the format (passed as first). See  [`errorf(java.lang.String,java.lang.Object)`](https://javadoc.io/static/io.quarkus/quarkus-core/2.7.2.Final/io/quarkus/logging/Log.html#errorf(java.lang.String,java.lang.Object)). Are you sure it would be printed when the format does not have a placeholder?